src/test/java/io/stargate/sgv2/jsonapi/fixtures/testdata/TestData.java

```diff
@@ -0,0 +1,55 @@
+package io.stargate.sgv2.jsonapi.fixtures.testdata;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class TestData {
+
+  protected final TestDataNames names;
+
+  private final Map<Class<?>, Object> cache = new HashMap<>();
+
+  public TestData() {
+    this(new TestDataNames());
+  }
+
+  protected TestData(TestDataNames names) {
+    this.names = names;
+  }
+
+  @SuppressWarnings("unchecked")
+  private <T> T getOrCache(Class<T> clazz) {
```

Comment:
Does this really make sense over eagerly constructed instances? It's not tons of code but... is there a real performance (or some other) problem  being solved?

Reply:
the idea was to share the same instances of things, such as the names